### PR TITLE
Allow setting extra component libraries at runtime

### LINKED ISSUE #6: Adding new components definitions at runtime
Is it possible to add new component definitions at runtime? As far as I can tell these are baked in at compile time from just the `sopc_components_*.xml` files in the source directory.

Although it's usually possible to add this information into the qsys description of an object with:

```tcl
set_module_assignment embeddedsw.dts.vendor my_vendor
set_module_assignment embeddedsw.dts.group my_name
set_module_assignment embeddedsw.dts.name my_name
```

This is more difficult to do for vendor supplied ip components, an example is the altera i2c master component.

For reference this is the component definition I'm compiling sopc2dts with
```xml
<SOPC2DTS_COMPONENTLIB vendor="altr">
        <S2DComponent classname="altera_avalon_i2c" group="i2c" compatDevice="i2c-master">
                <CompatibleVersion value="1.0"></CompatibleVersion>
  </S2DComponent>
</SOPC2DTS_COMPONENTLIB>
```

### PR DESCRIPTION
**Pull request by expipiplus1**
Closes #6

### CODE REVIEW

File: Sopc2DTS.java [hunk 1]
```diff
@@ -122,6 +125,7 @@ public Sopc2DTS() {
 		vOptions.add(new CommandLineOption("type", 		"t", outputType, 		true, false,"The type of output to generate", "{dtb,dtb-hex8,dtb-hex32,dtb-char-arr,dts,uboot,kernel}"));
 		vOptions.add(new CommandLineOption("bootargs", 	null,bootargs,	 		true, false,"Default kernel arguments for the \"chosen\" section of the DTS", "kernel-args"));
 		vOptions.add(new CommandLineOption("sopc-parameters", 	null,sopcParameters, true, false,"What sopc-parameters to include in DTS", "{node,cmacro,all}"));
+		vOptions.add(new CommandLineOption("extra-components",	"e", extraComponents, 	true, false,"Extra component descriptions (see source for examples) (can be used many times)", "sopc component xml file"));
```
Comment (wgoossens) [inline Sopc2DTS.java:129]: Just a nitpick...
Could you rename 'extra-components' to 'extra-componentlibs'?
Here and in the lines below

File: sopc2dts/lib/SopcComponentLib.java [hunk 2]
```diff
@@ -146,18 +146,23 @@ public void loadComponentLibsInWorkDir()
...
+		try {
+			Logger.logln("Loaded " + loadComponentLib(new InputSource(
+					new BufferedReader(new FileReader(filename))))
+					 + " components from " + filename, LogLevel.DEBUG);
+		} catch (FileNotFoundException e) {
+			Logger.logException(e);
+		}
+	}
+
 	public SopcComponentDescription getScdByClassName(String className)
```
Comment (wgoossens) [inline sopc2dts/lib/SopcComponentLib.java:166]: I like that refactured this to a new function instead of copy pasting.
Good Job.

**Review by wgoossens — CHANGES_REQUESTED** [review]
Small style change request, but all in all very good.

**Comment by expipiplus1** [comment]
Sorry this took so long to resolve!

**Comment by wgoossens** [comment]
Thank you for contributing!

**Pull request merged**